src/robotFactory.js

```diff
@@ -1,10 +1,94 @@
 'use strict';
 
-class BaseRobot {}
+class BaseRobot {
+  constructor(name, weight, { x = 0, y = 0 }, chipVersion) {
+    this.name = name;
+    this.weight = weight;
+    this.chipVersion = chipVersion;
 
-class FlyingRobot {}
+    this.coords = {
+      x,
+      y,
+    };
+  };
 
-class DeliveryDrone {}
+  getInfo() {
+    return `Robot: ${this.name}, `
+    + `Chip version: ${this.chipVersion}, `
+    + `Weight: ${this.weight}`;
+  }
+
+  goForward(step = 1) {
+    this.coords.y += step;
+  }
+
+  goBack(step = 1) {
+    this.coords.y -= step;
+  }
+
+  goRight(step = 1) {
+    this.coords.x += step;
+  }
+
+  goLeft(step = 1) {
+    this.coords.x -= step;
+  }
+}
+
+// const BaseRobot = new BaseRobot();
+
+class FlyingRobot extends BaseRobot {
+  constructor(name, weight, { x = 0, y = 0, z = 0 }, chipVersion) {
+    super(name, weight,
+      {
+        x,
+        y,
+      }, chipVersion);
+
+    this.coords = {
+      x,
+      y,
+      z,
+    };
```

Comment:
Можеш не дублювати код
```suggestion
    this.coords.z = z;
```

Reply:
В тебе ж є вже в **coords** **x** i **y**